src/ChipInput.js

```diff
@@ -291,13 +292,17 @@ class ChipInput extends React.Component {
   }
 
   handleKeyUp = (event) => {
-    if (this.props.newChipKeyCodes.indexOf(event.keyCode) < 0) {
+    if (this.props.newChipKeyCodes.indexOf(event.keyCode) < 0 && this.state.keyPressed) {
       this.setState({ inputValue: event.target.value })
     } else {
       this.clearInput()
     }
   }
 
+  handleKeyPress = (event) => {
+    this.setState({keyPressed: true})
```

Comment:
if `this.props.onKeyPress` is defined, it should be called so that this isnt a breaking change. But the same probably goes for `onKeyUp` in `handleKeyUp`. :thinking: